packages/core/src/extensions/extension-loader/file-system-provisioner-store/ensure-hashed-directory-for-extension.injectable.ts

```diff
@@ -24,14 +26,24 @@ const ensureHashedDirectoryForExtensionInjectable = getInjectable({
     const directoryForExtensionData = di.inject(directoryForExtensionDataInjectable);
     const ensureDirectory = di.inject(ensureDirInjectable);
     const getHash = di.inject(getHashInjectable);
+    const getPathToLegacyPackageJson = di.inject(getPathToLegacyPackageJsonInjectable);
 
     return async (extensionName, registeredExtensions) => {
-      const dirPath = await getOrInsertWithAsync(registeredExtensions, extensionName, async () => {
-        const salt = (await randomBytes(32)).toString("hex");
-        const hashedName = getHash(`${extensionName}/${salt}`);
+      let dirPath: string;
+
+      const legacyDirPath = getPathToLegacyPackageJson(extensionName);
+      const hashedDirectoryForLegacyDirPath = registeredExtensions.get(legacyDirPath);
+
+      if (hashedDirectoryForLegacyDirPath) {
+        dirPath = hashedDirectoryForLegacyDirPath;
+      } else {
+        dirPath = await getOrInsertWithAsync(registeredExtensions, extensionName, async () => {
+          const salt = (await randomBytes(32)).toString("hex");
```

Comment:
I don't think `randomBytes` should be `async` :)

Reply:
I think this is some kind of optimisation. 

> The crypto.randomBytes() method will not complete until there is sufficient entropy available. This should normally never take longer than a few milliseconds. The only time when generating the random bytes may conceivably block for a longer period of time is right after boot, when the whole system is still low on entropy.
> 
> This API uses libuv's threadpool, which can have surprising and negative performance implications for some applications; see the [UV_THREADPOOL_SIZE](https://nodejs.org/api/cli.html#uv_threadpool_sizesize) documentation for more information.
> 
> The asynchronous version of crypto.randomBytes() is carried out in a single threadpool request. To minimize threadpool task length variation, partition large randomBytes requests when doing so as part of fulfilling a client request.
> 

https://nodejs.org/api/crypto.html#cryptorandombytessize-callback

Reply:
anyway moved to the sync version of the random bytes